pubsub.go

```diff
@@ -209,10 +237,46 @@ func (p *PubsubValueStore) handleSubscription(sub *floodsub.Subscription, key st
 			if err != nil {
 				log.Warningf("PubsubResolve: error writing update for %s: %s", key, err)
 			}
+			p.notifyWatchers(key, msg.GetData())
+		}
+	}
+}
+
+func (p *PubsubValueStore) notifyWatchers(key string, data []byte) {
+	p.watchMux.RLock()
+	watchChannels, ok := p.watch[key]
+	if !ok {
+		p.watchMux.RUnlock()
+		return
+	}
+	watchChannels.mux.RLock()
+	p.watchMux.RUnlock()
+
+	defer watchChannels.mux.RUnlock()
+	for _, ch := range watchChannels.channels {
+		select {
+		case ch <- data:
```

Comment:
I'd consider:

1. Adding a buffer size of 1.
2. Replacing the buffered value in this case.

That is,

```go
select {
case ch <- data:
case <-ch:
    ch <- data
}
```

That way, the user always gets the latest value.